pkg/reconciler/gc/v2/gc.go

```diff
@@ -61,51 +64,67 @@ func Collect(
 		return a.After(b)
 	})
 
-	for _, rev := range revs[gcSkipOffset:] {
-		if strings.EqualFold(rev.ObjectMeta.Annotations[serving.RevisionPreservedAnnotationKey], "true") {
+	stale, nonactive := 0, 0
+	for _, rev := range revs {
+		if isRevisionActive(rev, config) {
+			continue
+		}
+		nonactive++
+		if isRevisionStale(cfg, rev, logger) {
+			stale++
+		}
+
+		if stale > min {
+			logger.Infof("Deleting stale revision %q", rev.ObjectMeta.Name)
+		} else if max != gc.Disabled && nonactive > max {
+			logger.Infof("Maximum(%d) reached. Deleting oldest non-active revision %q",
+				max, rev.ObjectMeta.Name)
+		} else {
 			continue
 		}
-		if isRevisionStale(ctx, rev, config) {
-			err := client.ServingV1().Revisions(rev.Namespace).Delete(rev.Name, &metav1.DeleteOptions{})
-			if err != nil {
-				logger.With(zap.Error(err)).Errorf("Failed to delete stale revision %q", rev.Name)
-				continue
-			}
+
+		if err := client.ServingV1().Revisions(rev.Namespace).Delete(
+			rev.Name, &metav1.DeleteOptions{}); err != nil {
+			logger.With(zap.Error(err)).Errorf("Failed to delete stale revision %q", rev.Name)
```

Comment:
```suggestion
			logger.With(zap.Error(err)).Error("Failed to GC revision: , rev.Name)
```
Not all of them are stale.